pkg/ingester/ingester_v2.go

```diff
@@ -1023,8 +1123,15 @@ func (i *Ingester) closeAllTSDB() {
 		}(userDB)
 	}
 
-	// Wait until all Close() completed
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		i.closeAllBackfillTSDBs()
+	}()
+
```

Comment:
I think we should do it after `i.userStatesMtx.Unlock()`. Also to further simplify the code, I would run `wg.Wait()` to wait until all user TSDBs are closed, and then I would call `closeAllBackfillTSDBs()` outside of any go routine.